GSoC_19/posts/websocket-passive-scan-using-scripts-with-zap.org

```diff
@@ -0,0 +1,845 @@
+   available by right clicking on them. **You should always remember that Right**
+   **Click in ZAP is your best friend.** By default, the folders under =Scripts=
+   are empty, except from the =WebSocket Passive Rules=. If you want to explore
+   more the scripting techniques and add some rules in ZAP I totally recommend
+   to install =Community Scripts= from =marketplace=. 
+   
+   **Hands On Section:**
+   Go to the script was created before. If it is not disabled, then disable it
+   using right click. Navigate over the code and find the following line: 
+   
+   #+BEGIN_SRC js
+   PRINT_RESULTS = false;
+   #+END_SRC
+   
+   make it =true=, enable and save the script. 
+
+   Passive scan scripts are running against all messages, as result we have not to
+   run them explicitly. So the only we have to do is to enable them, and when
+   some messages will be transmitted via the WebSocket connection the scripts will
+   be triggered.
+
+   **Hands On Section:** Connect to http://websocket.org/echo.html websocket
+   server as before, encode "Hello World" with a Base64 encoder
+   (SGVsbG8gV29ybGQ=) and send it. See what was printed in the scripts console.
+   
+   Then Disable your script, and find the following lines: 
+
+   #+BEGIN_SRC js
+   helper.newAlert()
+     .setRiskConfidence(RISK_INFO, CONFIDENCE_MEDIUM)
+     .setName("Base64 Disclosure in WebSocket message (script)")
+     .setDescription("A Base64-encoded string has been found in the websocket incoming message. Base64-encoded data may contain sensitive " +
+       "information such as usernames, passwords or cookies which should be further inspected. Decoded evidence: "
+       + decodedEvidence + ".")
+     .setSolution("Base64-encoding should not be used to store or send sensitive information.")
+     .setEvidence(evidence)
+     .raise();
+   #+END_SRC
+   and change the name of the alert. Send the Encoded "Hello World" once again
+   and then go to =Alert Panel= and see what happened. Finally, remove the
+   script using right click.
+
+   #+ATTR_HTML: :align center :width 900 height 400
+   #+CAPTION: Alert which is raised by MyBase64 script
+   [[../../images/pscan-tutorial/mybase64_alert.png]]
+
+   
+* WebSocket Passive Scan Scripts
+
+  By default ZAP comes with a various WebSocket Passive Scan Scripts. Default
+  Scripts are scanning passively the WebSocket Messages in order to disclosure
+  information which might be useful for the user while testing a Web
+  Application. As we saw before scripts are classified into two categories,
+  =Templates= and =Scripts=. Default scripts are located in both categories.
+  Those which are in =Scripts= category are those which actually running against
+  the WebSocket Messages. User can modify them any 
+  time they want or even remove them. By default those passive scan scripts are
+  enabled and they are located under =ZAP/scripts/scripts/websocketpassive=
```

Comment:
`<ZAP Home>` instead? The actual name depends on the OS https://github.com/zaproxy/zaproxy/wiki/FAQconfig